arm/vacuum_gripper.go

```diff
@@ -0,0 +1,159 @@
+package arm
+
+import (
+	"context"
+	"sync/atomic"
+
+	"github.com/golang/geo/r3"
+	"go.viam.com/rdk/components/arm"
+	"go.viam.com/rdk/components/gripper"
+	"go.viam.com/rdk/logging"
+	"go.viam.com/rdk/referenceframe"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/spatialmath"
+	"go.viam.com/utils"
+)
+
+// VacuumGripperModel model for the ufactory vacuum gripper.
+var VacuumGripperModel = family.WithModel("vacuum_gripper")
+
+// VacuumGripperConfig config for gripper.
+type VacuumGripperConfig struct {
+	Arm    string
+	Vacuum bool
+}
+
+// Validate validates the config.
+func (cfg *VacuumGripperConfig) Validate(path string) ([]string, error) {
+	if cfg.Arm == "" {
+		return nil, utils.NewConfigValidationFieldRequiredError(path, "board")
+	}
+
+	return []string{cfg.Arm}, nil
+}
+
+func init() {
+	resource.RegisterComponent(
+		gripper.API,
+		VacuumGripperModel,
+		resource.Registration[gripper.Gripper, *VacuumGripperConfig]{
+			Constructor: newVacuumGripper,
+		})
+}
+
+func newVacuumGripper(ctx context.Context, deps resource.Dependencies, config resource.Config, logger logging.Logger) (gripper.Gripper, error) {
+	newConf, err := resource.NativeConfig[*VacuumGripperConfig](config)
+	if err != nil {
+		return nil, err
+	}
+
+	g := &myVacuumGripper{
+		name:   config.ResourceName(),
+		mf:     referenceframe.NewSimpleModel("foo"),
+		conf:   newConf,
+		logger: logger,
+	}
+
+	g.arm, err = arm.FromDependencies(deps, newConf.Arm)
+	if err != nil {
+		return nil, err
+	}
+
+	return g, nil
+}
+
+type myVacuumGripper struct {
+	resource.AlwaysRebuild
+
+	name resource.Name
+	mf   referenceframe.Model
+
+	conf *VacuumGripperConfig
+
+	arm arm.Arm
+
+	isMoving atomic.Bool
+
+	logger logging.Logger
+}
+
+func (g *myVacuumGripper) Grab(ctx context.Context, extra map[string]interface{}) (bool, error) {
+	err := g.grabVacuum(ctx)
+	if err != nil {
+		return false, err
+	}
+
+	return true, nil
+}
+
+func (g *myVacuumGripper) Open(ctx context.Context, extra map[string]interface{}) error {
+	err := g.openVacuum(ctx)
```

Comment:
There's no need for this, just put the contents of `openVacuum` in this function. Same above with `Grab`